presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java

```diff
@@ -194,13 +194,22 @@ public void testMinMax()
     @Test
     public void testMinMaxBy()
     {
-        // TODO(spershin): Need to add use cases with non-numeric types when implemented.
         // We use filters to make queries deterministic.
         assertQuery("SELECT max_by(partkey, orderkey), max_by(quantity, orderkey), max_by(tax_as_real, orderkey) FROM lineitem where shipmode='MAIL'");
         assertQuery("SELECT min_by(partkey, orderkey), min_by(quantity, orderkey), min_by(tax_as_real, orderkey) FROM lineitem where shipmode='MAIL'");
 
         assertQuery("SELECT max_by(orderkey, extendedprice), max_by(orderkey, cast(extendedprice as REAL)) FROM lineitem");
         assertQuery("SELECT min_by(orderkey, extendedprice), min_by(orderkey, cast(extendedprice as REAL)) FROM lineitem where shipmode='MAIL'");
+
+        // 3 argument variant of max_by
+        assertQuery("SELECT max_by(orderkey, linenumber, 10), min_by(orderkey, linenumber, 10) FROM lineitem GROUP BY orderkey");
+
+        // Non-numeric arguments
+        assertQuery("SELECT max_by(row(orderkey, linenumber), linenumber, 10), min_by(row(orderkey, linenumber), linenumber, 10) FROM lineitem GROUP BY orderkey");
```

Comment:
It would be nice to include both group-by and global aggregations for completeness.